easy_rec/python/model/dssm.py

```diff
@@ -254,3 +259,23 @@ def get_outputs(self):
       return ['y', 'user_emb', 'item_emb']
     else:
       raise ValueError('invalid loss type: %s' % str(self._loss_type))
+
+  def build_output_dict(self):
+    output_dict = super(DSSM, self).build_output_dict()
+    output_dict['user_tower_feature'] = tf.reduce_join(tf.as_string(self.user_tower_feature), axis=-1, separator=',')
```

Comment:
this is the same as self._prediction_dict['user_emb']?

Reply:
No it's not. `user_emb` is the output of the user tower. `user_tower_feature` is the input of the user tower. We output the input of the user/item tower for debugging purpose.